cdap-api/src/main/java/co/cask/cdap/internal/workflow/ProgramWorkflowAction.java

```diff
@@ -40,9 +41,10 @@
 
   private final String name;
   private String programName;
-  private Callable<RuntimeContext> programRunner;
+  private Runnable programRunner;
   private SchedulableProgramType programType;
   private WorkflowContext context;
+  private WorkflowToken token;
```

Comment:
Don't you get a warning in your IDE about this? Mine warns me: token is never assigned. It also warns me that the context is assigned but never used.